yaml.go

```diff
@@ -85,30 +87,33 @@ type Marshaler interface {
 //
 // See the documentation of Marshal for the format of tags and a list of
 // supported tag options.
-func Unmarshal(in []byte, out any) (err error) {
-	return unmarshal(in, out, false)
+func Unmarshal(in []byte, out any, opts ...option.Option) (err error) {
+	config := option.NewConfig(opts...)
+	return unmarshal(in, out, config)
 }
 
 // A Decoder reads and decodes YAML values from an input stream.
 type Decoder struct {
-	parser      *parser
-	knownFields bool
+	parser *parser
+	config *option.Config
 }
 
 // NewDecoder returns a new decoder that reads from r.
 //
 // The decoder introduces its own buffering and may read
 // data from r beyond the YAML values requested.
-func NewDecoder(r io.Reader) *Decoder {
+func NewDecoder(r io.Reader, opts ...option.Option) *Decoder {
+	config := option.NewConfig(opts...)
 	return &Decoder{
 		parser: newParserFromReader(r),
+		config: config,
 	}
 }
 
 // KnownFields ensures that the keys in decoded mappings to
 // exist as fields in the struct being decoded into.
 func (dec *Decoder) KnownFields(enable bool) {
-	dec.knownFields = enable
+	dec.config.Apply(option.WithKnownFields(enable))
```

Comment:
You can't be sure the config field is set here.

You will have to init if missing. This should be done in a dedicated function of this struct that could be reused